frontend/src/scenes/onboarding/InviteSignup.tsx

```diff
@@ -0,0 +1,205 @@
+import { useActions, useValues } from 'kea'
+import React from 'react'
+import { hot } from 'react-hot-loader/root'
+import { inviteSignupLogic, ErrorCodes } from './inviteSignupLogic'
+import { SceneLoading } from 'lib/utils'
+import './InviteSignup.scss'
+import { StarryBackground } from 'lib/components/StarryBackground'
+import { userLogic } from 'scenes/userLogic'
+import { Button, Row, Col } from 'antd'
+import { ArrowLeftOutlined, ArrowRightOutlined } from '@ant-design/icons'
+import { router } from 'kea-router'
+import { PrevalidatedInvite } from '~/types'
+import { Link } from 'lib/components/Link'
+import { WhoAmI } from '~/layout/navigation/TopNavigation'
+import { LoginSignup } from './LoginSignup'
+
+const UTM_TAGS = 'utm_medium=in-product&utm_campaign=invite-signup'
+
+interface ErrorMessage {
+    title: string
+    detail: JSX.Element | string
+    actions: JSX.Element
+}
+
+function ErrorView(): JSX.Element | null {
+    const { error } = useValues(inviteSignupLogic)
+    const { user } = useValues(userLogic)
+    const { push } = useActions(router)
+
+    const HelperLinks = (
+        <>
+            <a className="plain-link" href="/">
+                App Home
+            </a>
+            <a
+                className="plain-link"
+                href={`https://posthog.com?${UTM_TAGS}&utm_message=invalid-invite`}
+                rel="noopener"
+            >
+                PostHog Website
+            </a>
+            <a
+                className="plain-link"
+                href={`https://posthog.com/slack?${UTM_TAGS}&utm_message=invalid-invite`}
+                rel="noopener"
+            >
+                Contact Us
+            </a>
+        </>
+    )
+
+    const BackToPostHog = (
+        <Button icon={<ArrowLeftOutlined />} block onClick={() => push('/')}>
+            Go back to PostHog
+        </Button>
+    )
```

Comment:
These should be `useMemo`ized (or in their own components), otherwise they are both rendered on every `ErrorView` render (and then one is always discarded).

Reply:
I assume you mean `ErrorMessages` as the other ones are static components?

Reply:
Actually `ErrorMessages` being less static makes it less important to memoize that. Would be good to memoize it too, but that's just a bit more work in figuring out correct dependencies. `HelperLinks` and `BackToPostHog` have almost no dependencies though AND they are rerendered needlessly all the more because of that – `const … = <…>…</…>` expressions are ran anew on _every_ component render.

Reply:
Moving this to functional components for code clarity as discussed on Slack.